tests/components/private_ble_device/test_coordinator.py

```diff
@@ -0,0 +1,158 @@
+"""Test the coordinator."""
+
+from home_assistant_bluetooth import BluetoothServiceInfoBleak
+
+from homeassistant.components.bluetooth.models import BluetoothChange
+from homeassistant.components.private_ble_device.coordinator import (
+    async_get_coordinator,
+)
```

Comment:
https://developers.home-assistant.io/docs/development_testing?_highlight=test#writing-tests-for-integrations

I'm pretty sure this counts as interacting with any integration details in tests of integrations.

You could steal some code from the ibeacon coordinator tests

Reply:
Ugh that's awkward. I was careful to avoid poking internals because of that rule. I can move some of the checks into the device tracker tests and not lose coverage. But I just can't get full coverage through the entity tests. The entity tests always use both async_track_* APIs and I really want to assert those APIs aren't accidentally coupled to each other in some way.

I suppose I could make a fake entity that only uses one API? I don't know if that's allowed either though (what would it's domain be).

Reply:
Ok, dropped that test file entirely and only had 2 or 3 uncovered lines. One was because of removing sensor.py. One was ^ but i've managed to avoid it.